`w` signature is not correctly reporting errors when the properties API is not met.

**Bug** <!-- delete as appropriate -->

The typing for the properties argument of `w` are not being respected, meaning that errors are not being reported when the argument does not match the properties API.

Package Version: <!-- package version -->

**Code**
```ts
interface Properties extends WidgetProperties {
    mand: boolean;
    key: string;
}

class Widget extends WidgetBase<Properties> {
}

w(Widget, { key: 'key' })
```

**Expected behaviour:**

`error TS2345: Argument of type '{ key: string; }' is not assignable to parameter of type 'DNode[] | Readonly<Properties> | undefined'.
  Type '{ key: string; }' is not assignable to type 'Readonly<Properties>'.
    Property 'mand' is missing in type '{ key: string; }'.`

**Actual behaviour:**

No Errors.

--- a/src/d.ts
+++ b/src/d.ts
@@ -70,19 +70,7 @@ export const registry = new WidgetRegistry();
 /**
  * Wrapper function for calls to create a widget.
  */
-export function w<P extends WidgetProperties>(widgetConstructor: WidgetBaseConstructor<P> | string, properties: P, children?: DNode[]): WNode;
-export function w<P extends WidgetProperties>(widgetConstructor: WidgetBaseConstructor<P> | string, children: DNode[]): WNode;
-export function w<P extends WidgetProperties>(widgetConstructor: WidgetBaseConstructor<P> | string): WNode;
-export function w<P extends WidgetProperties>(widgetConstructor: WidgetBaseConstructor<P> | string, propertiesOrChildren: P | DNode[] = <P> {}, children: DNode[] = []): WNode {
-		let properties: P;
-
-	if (Array.isArray(propertiesOrChildren)) {
-		children = propertiesOrChildren;
-		properties = <P> {};
-	}
-	else {
-		properties = propertiesOrChildren;
-	}
+export function w<P extends WidgetProperties>(widgetConstructor: WidgetBaseConstructor<P> | string, properties: WidgetProperties & P, children: DNode[] = []): WNode {
 
 	return {
 		children,

--- a/tests/unit/d.ts
+++ b/tests/unit/d.ts
@@ -1,7 +1,7 @@
 import * as registerSuite from 'intern!object';
 import * as assert from 'intern/chai!assert';
 import { assign } from '@dojo/core/lang';
-import { DNode, HNode } from '../../src/interfaces';
+import { DNode, HNode, WidgetProperties } from '../../src/interfaces';
 import { WidgetBase } from '../../src/WidgetBase';
 import { v, w, decorate, registry, WNODE, HNODE, isWNode, isHNode } from '../../src/d';
 import WidgetRegistry from './../../src/WidgetRegistry';
@@ -12,7 +12,11 @@ class TestFactoryRegistry extends WidgetRegistry {
 	}
 }
 
-class TestWidget extends WidgetBase<any> {
+interface TestProperties extends WidgetProperties {
+	mand: boolean;
+}
+
+class TestWidget extends WidgetBase<TestProperties> {
 	render() {
 		return v('outernode', { type: 'mytype' }, [
 			v('child-one'),
@@ -33,23 +37,6 @@ registerSuite({
 		assert.isObject(registry);
 	},
 	w: {
-		'create WNode wrapper with constructor'() {
-			const dNode = w(WidgetBase);
-			assert.deepEqual(dNode.widgetConstructor, WidgetBase);
-			assert.deepEqual(dNode.properties, {});
-			assert.equal(dNode.type, WNODE);
-			assert.isTrue(isWNode(dNode));
-			assert.isFalse(isHNode(dNode));
-		},
-		'create WNode wrapper with label'() {
-			const dNode = w('my-widget');
-
-			assert.equal(dNode.type, WNODE);
-			assert.deepEqual(dNode.widgetConstructor, 'my-widget');
-			assert.deepEqual(dNode.properties, {});
-			assert.isTrue(isWNode(dNode));
-			assert.isFalse(isHNode(dNode));
-		},
 		'create WNode wrapper using constructor with properties'() {
 			const properties: any = { id: 'id', classes: [ 'world' ] };
 			const dNode = w(WidgetBase, properties);
@@ -70,44 +57,24 @@ registerSuite({
 			assert.isTrue(isWNode(dNode));
 			assert.isFalse(isHNode(dNode));
 		},
-		'create WNode wrapper using constructor with children'() {
-			const properties: any = { id: 'id', classes: [ 'world' ] };
-			const dNode = w(WidgetBase, [ w(WidgetBase, properties) ]);
-
-			assert.equal(dNode.type, WNODE);
-			assert.deepEqual(dNode.widgetConstructor, WidgetBase);
-			assert.lengthOf(dNode.children, 1);
-			assert.isTrue(isWNode(dNode));
-			assert.isFalse(isHNode(dNode));
-		},
-		'create WNode wrapper using label with children'() {
-			const properties: any = { id: 'id', classes: [ 'world' ] };
-			const dNode = w('my-widget', [ w(WidgetBase, properties) ]);
-
-			assert.equal(dNode.type, WNODE);
-			assert.deepEqual(dNode.widgetConstructor, 'my-widget');
-			assert.lengthOf(dNode.children, 1);
-			assert.isTrue(isWNode(dNode));
-			assert.isFalse(isHNode(dNode));
-		},
 		'create WNode wrapper using constructor with properties and children'() {
-			const properties: any = { id: 'id', classes: [ 'world' ] };
-			const dNode = w(WidgetBase, properties, [ w(WidgetBase, properties) ]);
+			const properties = { mand: true };
+			const dNode = w(TestWidget, properties, [ w(WidgetBase, properties) ]);
 
 			assert.equal(dNode.type, WNODE);
-			assert.deepEqual(dNode.widgetConstructor, WidgetBase);
-			assert.deepEqual(dNode.properties, { id: 'id', classes: [ 'world' ]});
+			assert.deepEqual(dNode.widgetConstructor, TestWidget);
+			assert.deepEqual(dNode.properties, { mand: true });
 			assert.lengthOf(dNode.children, 1);
 			assert.isTrue(isWNode(dNode));
 			assert.isFalse(isHNode(dNode));
 		},
 		'create WNode wrapper using label with properties and children'() {
-			const properties: any = { id: 'id', classes: [ 'world' ] };
-			const dNode = w('my-widget', properties, [ w(WidgetBase, properties) ]);
+			const properties = { mand: false };
+			const dNode = w<TestProperties>('my-widget', properties, [ w(WidgetBase, properties) ]);
 
 			assert.equal(dNode.type, WNODE);
 			assert.deepEqual(dNode.widgetConstructor, 'my-widget');
-			assert.deepEqual(dNode.properties, { id: 'id', classes: [ 'world' ]});
+			assert.deepEqual(dNode.properties, { mand: false });
 			assert.lengthOf(dNode.children, 1);
 			assert.isTrue(isWNode(dNode));
 			assert.isFalse(isHNode(dNode));

